google/cloud/forseti/scanner/scanner_builder.py

```diff
@@ -51,60 +55,87 @@ def build(self):
-                module_path = 'google.cloud.forseti.scanner.scanners.{}'
-
-                requirements_map = scanner_requirements_map.REQUIREMENTS_MAP
-                if not scanner.get('name') in requirements_map:
-                    log_message = (
-                        'Configured scanner is undefined '
-                        'in scanner requirements map : %s')
-                    LOGGER.error(log_message, scanner.get('name'))
-                    continue
-
-                module_name = module_path.format(
-                    scanner_requirements_map.REQUIREMENTS_MAP
-                    .get(scanner.get('name'))
-                    .get('module_name'))
-
-                try:
-                    module = importlib.import_module(module_name)
-                except (ImportError, TypeError, ValueError):
-                    LOGGER.exception('Unable to import %s\n', module_name)
-                    continue
-
-                class_name = (
-                    scanner_requirements_map.REQUIREMENTS_MAP
-                    .get(scanner.get('name'))
-                    .get('class_name'))
-                try:
-                    scanner_class = getattr(module, class_name)
-                except AttributeError:
-                    LOGGER.exception('Unable to instantiate %s', class_name)
-                    continue
-
-                # Simple way to find the path to folders directory no matter
-                # where forseti runs.
-                rules_path = self.scanner_configs.get('rules_path')
-                if rules_path is None:
-                    scanner_path = inspect.getfile(scanner_class)
-                    rules_path = scanner_path.split('/google/cloud/forseti')[0]
-                    rules_path += '/rules'
-
-                rules_filename = (scanner_requirements_map.REQUIREMENTS_MAP
-                                  .get(scanner.get('name'))
-                                  .get('rules_filename'))
-                rules = '{}/{}'.format(rules_path, rules_filename)
-                LOGGER.info('Initializing the rules engine:\nUsing rules: %s',
-                            rules)
-
-                scanner = scanner_class(self.global_configs,
-                                        self.scanner_configs,
-                                        self.service_config,
-                                        self.model_name,
-                                        self.snapshot_timestamp,
-                                        rules)
-                runnable_scanners.append(scanner)
+        # please note that the scanner name passed in CLI must match the
+        # corresponding key in this dictionary
+        long_running_scanner_factory = {
+            'external_project_access_scanner': (
+                external_project_access_scanner.ExternalProjectAccessScanner,
+                'external_project_access_rules.yaml')
+        }
+        if self.scanner_name in long_running_scanner_factory:
```

Comment:
Hmmm.  I think that we should check if `self.scanner_name` is populated (i.e. not None) like below.  Otherwise, if someone makes a typo of a scanner name, all the other scanners will run.
```
if self.scanner_name:
    if self.scanner_name in long_running_scanner_factory:
else
    for scanner in self.scanner_configs.get('scanners'):
```

Reply:
this is actually handled in [the other PR](https://github.com/GoogleCloudPlatform/forseti-security/pull/2324/files#diff-b3ab31b8737b069d164ab56066436a0b). Well I was just having some issue making that PR specific to one file...